drivers/mspi/mspi_nrfe.c

```diff
@@ -449,27 +463,38 @@ static int api_get_channel_status(const struct device *dev, uint8_t ch)
 static int xfer_packet(struct mspi_xfer_packet *packet, uint32_t timeout)
 {
 	int rc;
+	nrfe_mspi_opcode_t opcode = (packet->dir == MSPI_RX) ? NRFE_MSPI_TXRX : NRFE_MSPI_TX;
+
+#ifdef CONFIG_MSPI_NRFE_IPC_NO_COPY
+	/* Check for alignment problems. */
+	uint32_t len = ((uint32_t)packet->data_buf) % sizeof(uint32_t) != 0
+			       ? sizeof(nrfe_mspi_xfer_packet_msg_t) + packet->num_bytes
+			       : sizeof(nrfe_mspi_xfer_packet_msg_t);
+#else
 	uint32_t len = sizeof(nrfe_mspi_xfer_packet_msg_t) + packet->num_bytes;
+#endif
 	uint8_t buffer[len];
```

Comment:
This buffer should be word aligned.
Also, it really should not be a local variable length array, but that can be fixed separately.